### PR TITLE
KYLIN-4698 Delete segment storage path after merging segment, deleting segment and droping cube

### PR DESCRIPTION
**Pull request by zzcclp**
## Proposed changes

Includes KYLIN-4660, KYLIN-4698, KYLIN-4699, and cherry-pick #1215 (KYLIN-2971)

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [ ] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java [hunk 1]
```diff
@@ -307,6 +307,7 @@ public String getHdfsWorkingDirectory() {
         String metaId = getMetadataUrlPrefix().replace(':', '-');
         //transform relative path for local metadata
         if (metaId.startsWith("../")) {
+            metaId = metaId.replace("../", "");
```
Comment (hit-lacus) [inline core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java:310]: This replace action looks confusing.

File: kylin-spark-project/kylin-spark-engine/src/test/java/org/apache/kylin/engine/spark/LocalWithSparkSessionTest.java [hunk 2]
```diff
@@ -191,7 +201,15 @@ protected ExecutableState mergeSegments(String cubeName, long start, long end, b
         CubeSegment mergeSegment = cubeMgr.mergeSegments(cube, new SegmentRange.TSRange(start, end), null, force);
         NSparkMergingJob mergeJob = NSparkMergingJob.merge(mergeSegment,  "ADMIN");
         execMgr.addJob(mergeJob);
-        return wait(mergeJob);
```
Comment (hit-lacus) [inline kylin-spark-project/kylin-spark-engine/src/test/java/org/apache/kylin/engine/spark/LocalWithSparkSessionTest.java:194]: Why remove `wait` method?

File: server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.apache.spark.sql.SparderContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+/**
+ * Created by zhangzc on 8/26/20.
```
Comment (hit-lacus) [inline server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java:33]: Please not leave your name.

File: server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ */
+public class InitialSparkerContext implements InitializingBean {
```
Comment (hit-lacus) [inline server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java:35]: `Sparker` or `Sparder`?

File: server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        runInitialSparder();
+    }
+
+    private void runInitialSparder() {
+        logger.info("Spark is starting.....");
+        SparderContext.init();
+        final String kylinHome = StringUtils.defaultIfBlank(KylinConfig.getKylinHome(), "./");
+        final File appidFile = Paths.get(kylinHome, "sparkappid").toFile();
```
Comment (hit-lacus) [inline server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java:48]: If current spark context is killed and restarted, will new applicationId update here?

File: server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java (hunk 4 above)
Comment (zzcclp) [reply server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java]: Rename to InitialSparderContext

File: server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java (hunk 5 above)
Comment (zzcclp) [reply server-base/src/main/java/org/apache/kylin/rest/init/InitialSparkerContext.java]: Done

**Review by hit-lacus — APPROVED** [review]
LGTM, thank you @zzcclp .

**Pull request merged**